### ISSUE TITLE
[BUG] [Linter]  T038 false positive: {% endif %}/{% endfor %} misreported as unmatched when `---` appears both outside and inside a conditional   block

### ISSUE THREAD

**Issue by tutomun** [djlint/djLint#2429]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)
  
## System Info

- OS: Windows 11
- Python Version: 3.13.9
- djLint Version: 1.44.2
- template language: jinja

## Issue

`T038` ("End tag has no matching block tag") is incorrectly raised on a correctly-balanced
`{% if %}...{% endif %}` / `{% for %}...{% endfor %}` pair. It only happens when a plain YAML
document separator (`---`) appears **both** somewhere before the Jinja block **and** inside a
conditionally-rendered part of that block. Neither `---` alone (outside only, or inside only)
triggers it — it's specifically the combination.

This surfaced in a real Ansible Jinja2 template (a Kubernetes YAML manifest generator) where a
`{% for %}` loop conditionally emits a new YAML document (`---` + content) per iteration — a
fairly common pattern for generating multi-document Kubernetes manifests from a template.

Worked around locally by wrapping the block in `{# djlint:off #}` / `{# djlint:on #}`.

## How To Reproduce

`repro.j2`:
```jinja
---
key: value
{% for x in [1, 2] %}
{% if x %}
---
key: value
{% endif %}
{% endfor %}

Run:
djlint repro.j2 --lint --configuration .djlintrc

Expected: no errors — every {% %} tag is correctly paired.

Actual:
T038 7:0 End tag has no matching block tag. {% endif %}
T038 8:0 End tag has no matching block tag. {% endfor %}
Removing either --- (the one before the Jinja block, or the one inside the {% if %}) makes
the false positive disappear — both are individually harmless, only the combination triggers it.

Contents of djlint.toml/.djlintrc/pyproject.toml [tool.djlint]

```json
{
  "profile": "jinja",
  "extension": "j2"
}
```
